### PR TITLE
Various K4A Viewer fixes

### LINKED ISSUE #419: OpenGL error: 'gl_FragColor' undeclared when showing point cloud in k4aviewer
**Describe the bug**
On a fresh installation of Ubuntu 18.04 with the default graphics drivers, switching to the point cloud view in K4AViewer results in a 'gl_FragColor' undefined error.
The current usage is deprecated and should be declared as an `out` variable in the shader.

**To Reproduce**
1. Install k4a debian packages on a fresh Ubuntu 18.04 installation.
2. Open a sample recording (I used OFFICE_SAMPLE_WFOV_UNBINNED.mkv)
3. Switch to the 3D / point cloud view.

**Expected behavior**
The point cloud view should be displayed.

**Logs**
```
terminate called after throwing an instance of 'std::logic_error'
  what():  Shader compilation error: 
0:15(2): error: `gl_FragColor' undeclared
0:15(2): error: value of type vec4 cannot be assigned to variable of type error
```

**Desktop (please complete the following information):**
 - OS with Version: Ubuntu 18.04 with latest Nouveau driver
 - SDK Version:  1.1.0
 - Firmware version:  N/A, recording file was used.

### PR DESCRIPTION
**Pull request by xthexder**
### Description of the changes:
- Fix IMU graph label being delayed #504 
- Fix point cloud fragment shader deprecated usage #419
- Fix playback of PASSIVE_IR recordings #424 
- Add better error checking for failed window creation #453 

### Before submitting a Pull Request:
- [x] I reviewed [CONTRIBUTING.md](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/CONTRIBUTING.md)
- [x] I [built my changes](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/building.md) locally
- [ ] I ran the [unit tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md)
- [ ] I ran the [functional tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md) with a device
- [ ] I ran the [performance tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md) with a device

### I tested changes on: 
- [x] Windows
- [ ] Linux

### CODE REVIEW

File: tools/k4aviewer/k4aimugraph.cpp [hunk 1]
```diff
@@ -121,7 +121,8 @@ void K4AImuGraph::PlotGraph(const char *name,
     std::stringstream nameBuilder;
     nameBuilder << "##" << name;
 
-    const float currentData = graphData[static_cast<size_t>(graphFrontIdx)].v[offset];
+    const float currentData =
+        graphData[(static_cast<size_t>(graphFrontIdx) + graphData.size() - 1) % graphData.size()].v[offset];
```
Comment (xthexder) [inline tools/k4aviewer/k4aimugraph.cpp]: @williampMSFT I just assumed this was a ring buffer, and it seems to work, but I couldn't really find any docs on the `ImGui::PlotLines` function. Hopefully you can tell me if my assumptions were correct.

Comment (williampMSFT) [reply tools/k4aviewer/k4aimugraph.cpp]: Yep, it's a ring buffer.

---
In reply to: [303670547](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/505#discussion_r303670547) [](ancestors = 303670547)

**Comment by williampMSFT** [comment]
Thanks for doing this :)

**Comment by xthexder** [comment]
I just finished testing this on linux. Unfortunately there are further issues preventing the point cloud viewer from running on linux.
I will open up a new issue after this is merged, since it's not related to k4aviewer.

**Pull request merged**